src/mongoNotes.js

```diff
@@ -0,0 +1,306 @@
+  const noteItemProjection = {
+    projection: {
+      _id: 0,
+      id: {$toString: '$_id'},
+      blocksInteraction: 1,
+      componentStyle: 1,
+      context: 1,
+      endDate: 1,
+      isDismissible: 1,
+      messageStyle: 1,
+      messageText: 1
+    }
+  };
+
+
+  return {addNoteItem, getNoteItem, getNoteItems, getNoteItemsForApp, removeNoteItem, removeNoteItemsByMessageStyle};
+
+
+  //-----------------------------------------------------------------------------
+  // ADD SERVER NOTIFICATIONS
+  //-----------------------------------------------------------------------------
+
+  /**
+   * Add note item to collection
+   * @param {Object} noteItem contains note item data:
+   * {
+   *   blocksInteraction: false,
+   *   componentStyle: "banner",
+   *   context: ["artikkelit", "muuntaja"],
+   *   endDate: new Date("2024-12-30T12:30:15.002"),
+   *   isDismissible: true,
+   *   messageStyle: "info",
+   *   messageText: "Test server notification message",
+   *  }
+   * @returns {void}
+   */
+  async function addNoteItem(noteItem) {
+    logger.info(`MongoNotes: Adding one note item ${JSON.stringify(noteItem)}`);
+
+    if (isNotObject(noteItem)) {
+      logger.debug('MongoNotes: NoteItem parameter is not object');
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, 'NoteItem data is not valid');
+    }
+
+    const newNoteItem = {
+      blocksInteraction: validate(noteItem.blocksInteraction, isBoolean),
+      componentStyle: validate(noteItem.componentStyle, isValidComponentStyle),
+      context: validate(noteItem.context, isValidContext),
+      endDate: validate(noteItem.endDate, isValidEndDate),
+      isDismissible: validate(noteItem.isDismissible, isBoolean),
+      messageStyle: validate(noteItem.messageStyle, isValidMessageStyle),
+      messageText: validate(noteItem.messageText, isValidMessageText)
+    };
+
+    if (hasUndefinedProperty(newNoteItem)) {
+      logger.debug('MongoNotes: NoteItem data did not pass validation');
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, 'NoteItem data is not valid');
+    }
+
+    // '_id' property is automatically generated by mongo node driver
+    // and added to noteItem before insert to collection.
+    // type of '_id' is ObjectId.
+    const result = await db.collection(collection).insertOne(newNoteItem);
+
+    if (result.acknowledged) {
+      logger.info(`New ${noteItem.componentStyle} note item added with ${noteItem.messageStyle} message: ${noteItem.messageText}`);
+
+      const {_id: _, ...rest} = newNoteItem;
+      return {id: result.insertedId.toString(), ...rest};
+    }
+
+    throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, `MongoNotes had error while adding a note ${noteItem}`);
+
+
+    /**
+     * Validate property with validator function
+     * @param {*} property
+     * @param {function} validator
+     * @returns {(*|undefined)}
+     */
+    function validate(property, validator) {
```

Comment:
Is this validate wrapper nessessary? Could those validation functions be free? And maybe even in utils.js or validate.js file?

Reply:
Notes validation functions and variables moved to validateNote.js and updated import/export usage as such.

Reply:
Wrapper is now removed and only data validator functions are used. Not valid data handling is in the usage level not baked in to the wrapper. Easier to set not valid default values if so needed.